File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] nums) {
    int n= nums.length; int s=0;
    int e=n-1; 

    if(nums[s]<= nums[e]){
        return nums[s];
    }
    while(s<e){
        int mid=s+(e-s)/2;
        if(nums[mid]<nums[mid+1] && nums[mid]<nums[n-1]){
            e=mid;
        }
        else if(nums[mid]<nums[mid+1] && nums[mid]>nums[n-1]){
            s=mid+1;
        }
        else{
            return nums[mid+1];
        }
     }   return 0;
    }
}

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution(object):
    def findMin(self, nums):
        n = len(nums)
        s = 0
        e = n-1
        mid = s + (e-s) // 2
        
        
        while s < e:
            num = nums[mid]
            if nums[e] < num:
                s = mid + 1

            else :
                e = mid        
               
            mid = s + (e-s)//2


        return nums[s]